WordPress/Classes/ViewRelated/NUX/LoginViewController.swift

```diff
@@ -12,19 +15,27 @@ extension LoginWithLogoAndHelpViewController where Self: UIViewController {
         let imageView = UIImageView(image: image.imageWithTintColor(UIColor.white))
         navigationItem.titleView = imageView
     }
+
     func addHelpButtonToNavController() -> (UIButton, WPNUXHelpBadgeLabel) {
         let helpButtonMarginSpacerWidth = CGFloat(-8)
         let helpBadgeSize = CGSize(width: 12, height: 10)
         let helpButtonContainerFrame = CGRect(x: 0, y: 0, width: 44, height: 44)
 
-        NotificationCenter.default.addObserver(self, selector: #selector(NUXAbstractViewController.handleHelpshiftUnreadCountUpdated(_:)), name: NSNotification.Name.HelpshiftUnreadCountUpdated, object: nil)
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.HelpshiftUnreadCountUpdated, object: nil, queue: nil) { [weak self](notification) in
+            self?.handleHelpshiftUnreadCountUpdated(notification)
+        }
 
         let customView = UIView(frame: helpButtonContainerFrame)
 
         let helpButton = UIButton(type: .custom)
         helpButton.setTitle(NSLocalizedString("Help", comment: "Help button"), for: .normal)
         helpButton.setTitleColor(UIColor(white: 1.0, alpha: 0.4), for: .highlighted)
-        helpButton.addTarget(self, action: #selector(NUXAbstractViewController.handleHelpButtonTapped(_:)), for: .touchUpInside)
+        helpButton.on(.touchUpInside) { [weak self](control: UIControl) in
```

Comment:
If you replace `(control: UIControl)` with just `control in`, the UIButton type gets auto-inferred, and you can drop the guard below.

Reply:
Good catch! Thanks